de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java

```diff
@@ -439,6 +450,62 @@ public Response deleteSei(@PathParam("seiUuid") @ApiParam(value = "Uuid of the S
 				return createSyncErrorResponse(e.getMessage());
 			}
 		}
+
+		/** **/
+		@GET
+		@Path(CONCEPTS)
+		@Produces(MediaType.APPLICATION_JSON)
+		@ApiOperation(
+				produces = "application/json",
+				value = "Fetch a list of active Concepts",
+				httpMethod = "GET",
+				notes = "This service fetches the active Concepts")
+		@ApiResponses(value = { 
+				@ApiResponse(
+						code = HttpStatus.OK_200,
+						response = ServerConcept.class,
+						responseContainer = "List",
+						message = SUCCESSFUL_OPERATION),
+				@ApiResponse(
+						code = HttpStatus.INTERNAL_SERVER_ERROR_500, 
+						message = SYNC_ERROR)})
+		public Response getConcepts(@DefaultValue("true") @QueryParam(QP_ONLY_ACTIVE_CONCEPTS) boolean onlyActiveConcepts) {
+			try {
+				serverRepository.syncRepository();
+				
+				List<ServerConcept> pojos = new ArrayList<ServerConcept>();
+				List<Concept> concepts;
+				
+				if (onlyActiveConcepts) {
+					concepts = repository.getActiveConcepts();
+				} else {
+					// Get all concepts from the registry
+					concepts = new ArrayList<Concept>();
+					IExtensionRegistry registry = Platform.getExtensionRegistry();
+					String extensionPoint = ActiveConceptConfigurationElement.EXTENSION_POINT_ID_CONCEPT;
+					
+					// Get the concept configuration elements
+					IConfigurationElement[] configurationElementsArray = registry.getConfigurationElementsFor(extensionPoint);
+					List<IConfigurationElement> configurationElements = (List<IConfigurationElement>) Arrays.asList(configurationElementsArray);

```

Comment:
Why do we need to create this list here? I see below we just use it to loop over the elements.
I think we can also just for each loop over the array directly.

Reply:
Yeah makes no sense, I will change it